build_tools/fetch_jax_wheels.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+"""
+Fetch JAX wheels from a cloudfront url.
+"""
+import os, re, sys, fnmatch
+from urllib.parse import urljoin, urlparse
+from urllib.request import urlopen, urlretrieve
+
+wh = os.environ.get("WHEELHOUSE_DIR", "wheelhouse")
+pv, pi, af = os.environ.get("PY_VER"), os.environ.get("PACKAGE_INDEX_URL"), os.environ.get("AMDGPU_FAMILY")
+if not (pv and pi and af): sys.exit("Set PY_VER, PACKAGE_INDEX_URL, AMDGPU_FAMILY")
+
+# Build the CP tag and the base listing URL
+cp, base = pv.replace(".", ""), f"{pi.rstrip('/')}/{af.strip('/')}/"
+os.makedirs(wh, exist_ok=True)
+
+def links(u):
+    """Fetch a page and return absolute HREFs found on it. On failure, return empty list."""
+    try:
+        h = urlopen(u, timeout=30).read().decode("utf-8", "ignore")
+    except Exception:
+        return []
+    return [urljoin(u, x) for x in re.findall(r'href=[\'"]([^\'"]+)[\'"]', h, flags=re.I)]
```

Comment:
Why swallow the error and return empty? Probably should re-raise the Exception.

Also this needs a more descriptive function name.